site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+       `RouteRules` that use `AllowedRoutes`.
+    4. `AllowedRoutes` MUST be non-nil in a `RouteRule` to permit child `Routes`
+       to attach to that `RouteRule`.
+        1. Explanation: A parent `Route` will allow delegation to one or more
+           child `Routes` by specifying `AllowedRoutes` where `BackendRef` would
+           be used otherwise.
+    5. Multiple child routes may attach to a single `RouteRule`, as long as they
+       are distinct.
+3. Allow a Route’s `ParentRef` Kind to be the same type as the Route’s type
+    1. `Kind` is required when the `ParentRefs` is a route. `Kind` wil default to
+       `gateway` (current behavior)
+4. Add `Name` (type is [SectionName][sectionName]) to
+   [HTTPRouteRule][httpRouteRule] as optional
+    1. The `Name` field is expected to be added in the implementation of
+       [GEP-995 - Add Name to HTTPRouteRule and HTTPRouteMatch][gep-995]
+    2. Default value is nil
+    3. A child route may include a `SectionName` when its `ParentRef` is a route.
+        1. A child route “attaches" to the parent route’s `RouteRule` that matches
+           that `SectionName`. The child route must also match the criteria
+           specified in the `RouteRule’s AllowedRoutes`.
+    4. If the `ParentRefs` of a child route doesn’t include a `SectionName`, the
+       child route attaches to all of the Parent Route’s `RouteRules` having
+       `AllowedRoutes` criteria that the child route meets.
+5. A parent `Route` and any related child `Routes` must be the same type of
+   `Routes`.
+    1. Example: A child httpRoute can not point to a parent tcpRoute.
+6. A `Route` MUST NOT be a parent to a `Route` AND a child of another `Route`.
+    1. Explanation: This limitation avoids the need for specific
+       internal/terminal Route types, and the design of using a ParentRef from
+       the child to an AllowedRoutes slot on a parent allows a Route to know if
+       it is either a parent or child (or misconfigured as both) without
+       requiring the graph traversal that a design like ForwardTo may need to
+       avoid cycles.
+    2. If an implementation chooses to not enforce this limitation, that must be
+       documented along with how many tiers of parent/child route relationships
+       are allowed
+7. A Route’s `ParentRefs` can list more than one type of parent.
+    1. Example: A `HTTPRoute`’s ParentRefs can list one (or more) `Gateways` and
+       one (or more) `HTTPRoutes` at the same time.
+
+## Binding Parent and Child Routes
+
+Binding a child route to a parent route refers to the process of reconciling
+(i.e. combining) the two routes. Binding is treated as a single transaction for
+each Parent/Child route pair. Even when a child route attaches to multiple
+`RouteRules` in the parent route, the binding between the two routes is still a
+single transaction.
+
+The binding transaction has two stages; reconciliation and activation. The
```

Comment:
I am a bit confused by this.. it sounds like its describing an implementation. Can we rephrase it as a spec?

Reply:
An implementation may choose to use a controller to combine the routes and generated 'flat' routes - instead of implementing this at 'runtime'. Or may use tools in CI/CD to combine the routes and deploy flat routes. 

I agree a spec should not require or describe a particular implementation.